File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/AstralImprisonment.cs
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class AstralImprisonment : Affliction
    {
        public override string Name { get { return "Astral Imprisonment"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.Attacking, 
                };
            }
        }

        protected override void OnAttacking(ICharacter attacker, ICharacter target, ISpell spell, ref double damage, ref double modifier)
        {
            if (RollPercentage(10))
            {
                ApplyAffliction(target, "Mini Stun");
            }
        }
    }
}

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/lua/AstralImprisonment.lua
function GetName()
  return "Astral Imprisonment"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { Attacking }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == Attacking and Roll(0, 99) < 10 then
    param1:ApplyAffliction("Mini Stun")
  end
end
